snapcraft/cli/lifecycle.py

```diff
@@ -31,6 +40,9 @@ def _execute(command, parts, **kwargs):
     else:
         try:
             lifecycle.execute(command, project_options, parts)
+        except errors.SnapcraftUserPermissionError as user_error:
+            if user_error.user == 'root':
+                _reexec_as_root()
```

Comment:
build actually has the sudo calls in the build scripts of core itself now via https://github.com/snapcore/core/pull/51 ... what failed hard was prime even after this change ...